File: tfm-sgpfe-simulator-rneves/service_simulator/lib/modules/common/domain/entities/queue.dart
import 'package:equatable/equatable.dart';
import 'package:meta/meta.dart';

class Queue extends Equatable {
  final int id;
  final String letter;
  final String name;
  final String description;
  final int servicePostOfficeId;
  final int activeServers;
  final int type;
  final int maxAvailable;
  final bool tolerance;

  Queue(
      {@required this.id,
      @required this.description,
      @required this.letter,
      @required this.name,
      @required this.activeServers,
      @required this.type,
      @required this.maxAvailable,
      @required this.tolerance,
      @required this.servicePostOfficeId});

  @override
  List<Object> get props => [
        id,
        letter,
        name,
        description,
        servicePostOfficeId,
        activeServers,
        type,
        maxAvailable,
        tolerance
      ];
}

File: tfm-sgpfe-rneves/server/src/main/kotlin/isel/meic/tfm/fei/entities/Queue.kt
package isel.meic.tfm.fei.entities

/**
 * Instituto Superior de Engenharia de Lisboa
 * Project: Smart Queue
 * @author Ronilton Neves - 39643
 * @mentor: Paulo Pereira
 *
 * <Description>
 */
data class Queue(
    val id: Int,
    val name : String,
    val activeServers: Int,
    val letter: String,
    val description: String,
    val type: Int,
    val maxAvailable: Int,
    val servicePostOfficeId: Int,
    val tolerance: Boolean
)